azure/scope/cluster.go

```diff
@@ -539,13 +539,14 @@ func (s *ClusterScope) AzureBastion() *infrav1.AzureBastion {
 }
 
 // AzureBastionSpec returns the bastion spec.
-func (s *ClusterScope) AzureBastionSpec() azure.ResourceSpecGetter {
+func (s *ClusterScope) AzureBastionSpec() azure.ASOResourceSpecGetter[*asonetworkv1.BastionHost] {
```

Comment:
Is there a reason we want to return a list of pointers rather than just a list of objects, i.e. `azure.ASOResourceSpecGetter[asonetworkv1.BastionHost]`?

Reply:
It's a convention for implementing ASO resources with Generics, although I'm not sure about the specific reason. @nojnhuh might know!

https://gist.github.com/nojnhuh/a1e06bb6d98b0eacb51134480ea4acea#option-1-use-azureserviceasos-service

Reply:
The `[]` syntax is a little confusing here in that it's not for a list but for type parameters. We use the pointer type instead of values because all of the handy methods common to Kubernetes resource types in ASO's [`genruntime.MetaObject`](https://pkg.go.dev/github.com/Azure/azure-service-operator/v2@v2.3.0/pkg/genruntime#MetaObject) are implemented on the pointer type, e.g. https://pkg.go.dev/github.com/Azure/azure-service-operator/v2@v2.3.0/api/resources/v1api20200601#ResourceGroup.